src/components/Modal/PopupModal.js

```diff
@@ -27,27 +26,68 @@ const useStyles = makeStyles(theme => ({
   },
 }))
 
-export default function TransitionsModal() {
+export default function AgeVerificationModal() {
   const isBrowser = typeof window !== "undefined"
-
   const classes = useStyles()
-  const [open, setOpen] = React.useState(true)
-  const [sent, setSent] = React.useState(false)
 
-  const handleOpen = () => {
-    setOpen(true)
+  const [dob, setDob] = useState("")
+  const [ageVerified, setAgeVerified] = useState(false)
+  const [errorMessage, setErrorMessage] = useState("")
+  const [open, setOpen] = useState(true)
+
+  // Age threshold for verification (18 years old)
+  const ageThreshold = 18
+
+  // Function to calculate age from DOB
+  const calculateAge = dob => {
+    const birthDate = new Date(dob)
+    const today = new Date()
+    const age = today.getFullYear() - birthDate.getFullYear()
+    const month = today.getMonth() - birthDate.getMonth()
+
+    if (month < 0 || (month === 0 && today.getDate() < birthDate.getDate())) {
+      return age - 1
+    }
+    return age
+  }
+
+  // Handle form submission
+  const handleSubmit = e => {
+    e.preventDefault()
+
+    if (!dob) {
+      setErrorMessage("Please enter your date of birth.")
+      return
+    }
+
+    const age = calculateAge(dob)
+
+    if (age >= ageThreshold) {
+      setAgeVerified(true)
+      setErrorMessage("")
+      setOpen(false) // Close modal if age is valid
+      if (isBrowser) {
+        sessionStorage.setItem("ageVerified", "true") // Store verified status
+      }
+    } else {
+      setAgeVerified(false)
+      setErrorMessage(`You must be at least ${ageThreshold} years old.`)
+      if (isBrowser) {
+        window.location.href = "https://www.google.com"
```

Comment:
I don't think we should redirect to google here. Maybe we could add a simple page on the simmons site to redirect to, or just displaying the error message is enough? 🤷